keccak256/src/gates/pi.rs

```diff
@@ -12,66 +9,50 @@ use std::marker::PhantomData;
 
 #[derive(Clone, Debug)]
 pub struct PiConfig<F> {
-    q_enable: Selector,
     state: [Column<Advice>; 25],
     _marker: PhantomData<F>,
 }
 
 impl<F: FieldExt> PiConfig<F> {
-    pub const OFFSET: usize = 2;
```

Comment:
Please. Don't remove the OFFSET as it's required later for assembliyng the entire Keccak circuit.

Reply:
Sorry, I'll return them back.

Reply:
Oh, I realize it's part of the layouter refactor. 

Currently the circuit looks like this
```
  state = {
                // Apply pi outside circuit
                let out_state = KeccakFArith::pi(&state_to_biguint(state));
                let out_state = state_bigint_to_pallas(out_state);
                // assignment
                self.pi_config
                    .assign_state(region, offset, state, out_state)?
            };

            offset += PiConfig::OFFSET;
```
With the new interface, we can do
```
  state = self.pi_config.assign_state(mut layouter, state, out_state)?;
```
No need to track the region and the offset. The layouter find a new region for it.
What do you think?

Reply:
hmmm. 

So the proposal is to have each gate/config application within a different region then?

Reply:
Yes, it makes sense to let layouter decide where to put the state row. Counting offsets is like doing the layouter's job.